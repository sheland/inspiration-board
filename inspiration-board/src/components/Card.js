import React from 'react';
import PropTypes from 'prop-types';
import emoji from 'emoji-dictionary';

import './Card.css';

const InspirationCard = (props) => {
  return (
    <div className="card">
      <section className="card__content">
        <p className="card__content-text">{props.cardText}</p>

        {props.cardEmoji && (
          <p className="card__content-emoji">
            {emoji.getUnicode(props.cardEmoji)}
          </p>)}

          <button
            onClick={() => props.deleteCardCallback(props.cardId)}
            type="button"
            className="card__delete"
            > Delete </button>
          </section>
        </div>
      )
    }


    InspirationCard.propTypes = {
      cardText: PropTypes.string.isRequired,
    };

    export default InspirationCard;
