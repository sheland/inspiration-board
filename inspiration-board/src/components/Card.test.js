import React from 'react';
import Card from './Card';
import { shallow } from 'enzyme';

describe('card', ()=> {
  it('will match the Card Snapshot', ()=> {
    const wrapper = shallow(
      <Card
      key = {1}
      cardText="dog"
      cardEmoji="heart_eyes"
      cardId = {1}
      deleteCardCallback = {()=> {}}
      />);
      expect(wrapper).toMatchSnapshot();

    });
  });
