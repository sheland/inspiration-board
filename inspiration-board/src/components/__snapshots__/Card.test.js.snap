// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`card will match the Card Snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InspirationCard
    cardEmoji="heart_eyes"
    cardId={1}
    cardText="dog"
    deleteCardCallback={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <section
        className="card__content"
      >
        <p
          className="card__content-text"
        >
          dog
        </p>
        <p
          className="card__content-emoji"
        >
          üòç
        </p>
        <button
          className="card__delete"
          onClick={[Function]}
          type="button"
        >
           Delete 
        </button>
      </section>,
      "className": "card",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="card__content-text"
          >
            dog
          </p>,
          <p
            className="card__content-emoji"
          >
            üòç
          </p>,
          <button
            className="card__delete"
            onClick={[Function]}
            type="button"
          >
             Delete 
          </button>,
        ],
        "className": "card__content",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "dog",
            "className": "card__content-text",
          },
          "ref": null,
          "rendered": "dog",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "üòç",
            "className": "card__content-emoji",
          },
          "ref": null,
          "rendered": "üòç",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " Delete ",
            "className": "card__delete",
            "onClick": [Function],
            "type": "button",
          },
          "ref": null,
          "rendered": " Delete ",
          "type": "button",
        },
      ],
      "type": "section",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <section
          className="card__content"
        >
          <p
            className="card__content-text"
          >
            dog
          </p>
          <p
            className="card__content-emoji"
          >
            üòç
          </p>
          <button
            className="card__delete"
            onClick={[Function]}
            type="button"
          >
             Delete 
          </button>
        </section>,
        "className": "card",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="card__content-text"
            >
              dog
            </p>,
            <p
              className="card__content-emoji"
            >
              üòç
            </p>,
            <button
              className="card__delete"
              onClick={[Function]}
              type="button"
            >
               Delete 
            </button>,
          ],
          "className": "card__content",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "dog",
              "className": "card__content-text",
            },
            "ref": null,
            "rendered": "dog",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "üòç",
              "className": "card__content-emoji",
            },
            "ref": null,
            "rendered": "üòç",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " Delete ",
              "className": "card__delete",
              "onClick": [Function],
              "type": "button",
            },
            "ref": null,
            "rendered": " Delete ",
            "type": "button",
          },
        ],
        "type": "section",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
